core/objects.go

```diff
@@ -191,6 +193,26 @@ type ValidationRecord struct {
 	AddressUsed       net.IP   `json:"addressUsed"`
 }
 
+// AuthorizedKey represents a domain holder's authorization for a
+// specific account key to satisfy a specific challenge.
+type AuthorizedKey struct {
+	Token string           `json:"token"`
+	Key   *jose.JsonWebKey `json:"key,omitempty"`
+}
+
+// Match determines whether this AuthorizedKey object matches the given token and key
+func (ak AuthorizedKey) Match(token string, key *jose.JsonWebKey) bool {
+	if ak.Key == nil {
+		return false
+	}
+	return token == ak.Token && KeyDigestEquals(key.Key, ak.Key.Key)
```

Comment:
Please respond to my comment about why crypto/subtle is not needed here or make the changes.